asterisk/asterisk
# 18.19.0 issue 307

- Update for 18.17.0-rc1
- Update for 18.17.0
- Revert "pbx_ael: Global variables are not expanded."
- res_pjsip_pubsub: subscription cleanup changes
- Update for 18.17.1
- res_http_media_cache: Introduce options and customize
- main/iostream.c: fix build with libressl
- contrib: rc.archlinux.asterisk uses invalid redirect.
- app_senddtmf: Add SendFlash AMI action.
- res_agi: RECORD FILE plays 2 beeps.
- cli: increase channel column width
- configure: fix detection of re-entrant resolver functions
- make_version: Strip svn stuff and suppress ref HEAD errors
- app_dial: Fix DTMF not relayed to caller on unanswered calls.
- func_json: Fix JSON parsing issues.
- app_queue: periodic announcement configurable start time.
- format_sln: add .slin as supported file extension
- res_mixmonitor: MixMonitorMute by MixMonitor ID
- bridge_builtin_features: add beep via touch variable
- res_calendar: output busy state as part of show calendar.
- ael: Regenerate lexers and parsers.
- loader.c: Minor module key check simplification.
- test.c: Fix counting of tests and add 2 new tests
- app_queue: Fix minor xmldoc duplication and vagueness.
- voicemail.conf: Fix incorrect comment about #include.
- chan_pjsip: fix music on hold continues after INVITE with replaces
- install_prereq: Add Linux Mint support.
- res_pjsip_stir_shaken: Fix JSON field ordering and disallowed TN characters.
- pbx_dundi: Add PJSIP support.
- Revert "app_queue: periodic announcement configurable start time."
- pbx_dundi: Fix PJSIP endpoint configuration check.
- Initial GitHub Issue Templates
- Initial GitHub PRs
- .github: Remove unnecessary parameter in CherryPickTest
- .github: Update issue templates
- chan_dahdi: Add dialmode option for FXS lines. (#36)
- LICENSE: Update link to trademark policy.
- .github: Add cherry-pick test progress labels
- res_pjsip: mediasec: Add Security-Client headers after 401
- .github: Refactor CP progress and add new PR test progress
- .github: Remove separate set labels step from new PR
- .github: Fix reference to CHERRY_PICK_TESTING_IN_PROGRESS
- .github: Fix CherryPickTest to only run when it should
- core: Cleanup gerrit and JIRA references. (#40)
- cel: add local optimization begin event
- chan_pjsip: also return all codecs on empty re-INVITE for late offers
- .github: Add AsteriskReleaser
- Update for 18.18.0-rc1
- Set up new ChangeLogs directory
- Update for 18.18.0
- pjsip: Upgrade bundled version to pjproject 2.13.1
- apply_patches: Sort patch list before applying
- apply_patches: Use globbing instead of file/sort.
- Update for 18.18.1
- .github: Fix staleness check to only run on certain labels.
- .github: Tweak new feature language, and move feature requests elsewhere.
- .github: Tweak improvement issue type language.
- xml.c: Process XML Inclusions recursively.
- res_sorcery_memory_cache.c: Fix memory leak
- res_pjsip_pubsub.c: Use pjsip version for pending NOTIFY check. (#76)
- configure: fix test code to match gethostbyname_r prototype.
- .github: Add cherry-pick reminder to new PRs
- .github: Fix quoting in PROpenedOrUpdated
- .github: Don't add cherry-pick reminder if it's already present
- asterisk.c: Fix option warning for remote console.
- .github: Change title of AsteriskReleaser job
- chan_dahdi: Fix broken hidecallerid setting.
- chan_pjsip: Allow topology/session refreshes in early media state
- callerid: Allow specifying timezone for date/time.
- .github: Update field descriptions for AsteriskReleaser
- .github Ignore error when adding reviewrs to PR
- indications: logging changes
- .github: Fix issues with cherry-pick-reminder
- app_queue: Preserve reason for realtime queues
- res_stasis.c: Add new type 'sdp_label' for bridge creation.
- sig_analog: Add fuller Caller ID support.
- AMI: Add CoreShowChannelMap action.
- chan_dahdi: Fix Caller ID presentation for FXO ports.
- res_musiconhold: Add option to loop last file.
- configure: Makefile downloader enable follow redirects.
- sounds: Update download URL to use HTTPS.
- app_followme: fix issue with enable_callee_prompt=no (#88)
- ast-db-manage: Fix alembic branching error caused by #122.
- .github: Rework for merge approval
- build: Fix a few gcc 13 issues
- test_stasis_endpoints.c: Make channel_messages more stable
- AMI: Add parking position parameter to Park action
- tcptls: when disabling a server port, we should set the accept_fd to -1.
- res_crypto.c: Avoid using the non-portable ALLPERMS macro.
- test_statis_endpoints:  Fix channel_messages test again
- ast-db-manage: Synchronize revisions between comments and code.
- res_fax_spandsp.c: Clean up a spaces/tabs issue
- configure: Remove obsolete and deprecated constructs.
- res_crypto.c: Gracefully handle potential key filename truncation.
- pjsip_transport_events.c: Use %zu printf specifier for size_t.
- app_queue: Add force_longest_waiting_caller option.
- res_pjsip_session: Added new function calls to avoid ABI issues.
- res_pjsip_rfc3326: Prefer Q.850 cause code over SIP.
- app_voicemail_imap: Fix message count when IMAP server is unavailable
- .github: Allow PR submit checks to be re-run by label
- .github: Fix recheck label manipulation
- .github: Fix recheck label typos
- .github: Add recheck label test to additional jobs
- .github: Fix use of 'contains'
- .github: Remove result check from PROpenUpdateGateTests
- rest-api: Updates for new documentation site
- .github: Move publish docs to new file CreateDocs.yml
- .github: Back out triggering PROpenedOrUpdated by label
- utils: add lock timestamps for DEBUG_THREADS
- res_musiconhold: avoid moh state access on unlocked chan
- app_voicemail: fix imap compilation errors
- .github: Updates for AsteriskReleaser
- Update for 18.19.0-rc1
- doc: Remove obsolete CHANGES-staging and UPGRADE-staging
- app.h: Move declaration of ast_getdata_result before its first use
- Update for 18.19.0-rc2
- Update for 18.19.0
- channels: Avoid deadlocks when locking ao2_container *channels
